cmd/checkup-launcher.go

```diff
@@ -49,18 +49,33 @@ func main() {
 		}
 	}()
 
+	status := checkup.NewStatus()
+
+	status.SetStartTimestampToNow()
 	jobErr := workspace.StartAndWaitCheckupJob(clientset)
+	status.SetCompletionTimestampToNow()
 
 	checkupJob := workspace.Job()
 	if err := logCheckupJobLogs(clientset, checkupJob.Namespace); err != nil {
 		log.Printf("Failed to dump checkup job logs: %v", err)
 	}
 
 	if jobErr != nil {
-		log.Fatalf("Error occured while running checkup job: %v", jobErr)
+		status.SetFailureReason(jobErr.Error())
+		log.Printf("Error occured while running checkup job: %v", jobErr)
 	}
+
 	if isJobFailed(checkupJob) {
-		log.Fatalf("Checkup job completed with failure")
+		log.Printf("Checkup job completed with failure")
+	}
+
+	if err := workspace.RetrieveCheckupStatus(clientset, status); err != nil {
+		log.Printf("Failed to retrive checkup status: %v\n", err.Error())
+	}
+
+	checkup.AppendCheckupStatusToConfigMap(configMap, status)
+	if err := setStatusToUserConfigMap(clientset, cmNamespace, configMap); err != nil {
```

Comment:
Please consider renaming `configMap` `cmNamespace` and `cmName` to a name that will emphasize 
that this the framework config ConfigMap (and not the intermediate one the checkup updates) 
maybe something like `frameworkConfigMap`.

Reply:
If we change it here, we also have to change it where we get the spec.